src/core/reference/src/runtime/reference/utils/nms_common.cpp

```diff
@@ -18,11 +18,29 @@ void nms_common_postprocessing(void* prois,
                                const ngraph::element::Type& output_type,
                                const std::vector<float>& selected_outputs,
                                const std::vector<int64_t>& selected_indices,
-                               const std::vector<int64_t>& valid_outputs) {
+                               const std::vector<int64_t>& valid_outputs,
+                               const ngraph::element::Type& selected_outputs_type) {
     int64_t total_num = std::accumulate(valid_outputs.begin(), valid_outputs.end(), 0);
 
-    float* ptr = static_cast<float*>(prois);
-    memcpy(ptr, selected_outputs.data(), total_num * sizeof(float) * 6);
+    switch (selected_outputs_type) {
+    case element::Type_t::bf16: {
+        bfloat16* ptr = static_cast<bfloat16*>(prois);
+        for (auto i = 0; i < total_num * 6; ++i) {
+            ptr[i] = bfloat16(selected_outputs[i]);
+        }
+    } break;
+    case element::Type_t::f16: {
+        float16* ptr = static_cast<float16*>(prois);
+        for (auto i = 0; i < total_num * 6; ++i) {
+            ptr[i] = float16(selected_outputs[i]);
+        }
+    } break;
+    case element::Type_t::f32: {
+        float* ptr = static_cast<float*>(prois);
+        memcpy(ptr, selected_outputs.data(), total_num * sizeof(float) * 6);
+    } break;
+    default:;
```

Comment:
Should we prohibit unsupported element types?

Reply:
Updated.